src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,91 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData = new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        resize(size);
+        elementData[size] = value;
+        size += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        int indexDesPos = index + 1;
+        if (index == size) {
+            indexDesPos = index;
+        }
+        size += 1;
+        resize(size);
+        checkIndex(index);
+        System.arraycopy(elementData,index,elementData,indexDesPos,size - indexDesPos);
+        elementData[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Object oldValue = elementData[index];
+        size = size - 1;
+        System.arraycopy(elementData,index + 1,elementData,index,size - index);
+        return (T) oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < elementData.length; i++) {
+            if (elementData[i] == element
+                    || (elementData[i] != null && elementData[i].equals(element))) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException();
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("index is out of range");
+        }
+    }
+
+    private void resize(int size) {
```

Comment:
```suggestion
    private void resize() {
```
у нас є доступ до змінною класу, можна не очікува її на вхід в метод